src/Data/Repositories/Repository.cs

```diff
@@ -18,115 +15,38 @@ protected Repository(ApplicationContext context)
             Set = context.Set<TEntity>();
         }
 
-        public virtual async Task<int> Count(FilterRule<TEntity, TKey> filterRule = null)
+        public async Task<TEntity> GetAsync(TKey id)
         {
-            var query = Set.AsQueryable();
-
-            if (filterRule == null)
-                return await query.CountAsync();
-
-            return await FilterQuery(query, filterRule).CountAsync();
+            return await Set.AsQueryable().FirstOrDefaultAsync(entity => entity.Id.Equals(id));
```

Comment:
I think that we don't need AsQueryable here

Reply:
Done